src/components/NavBar/NavBar.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Button from '@material-ui/core/Button';
+import { useNavBarStyles } from "./NavBar.style"
+import { useState } from 'react';
+import { Redirect, useHistory } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { setLikedPhotosId, setToken } from '../../actions';
+import { loginUrl } from '../../app/unsplash';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import { IconButton } from '@material-ui/core';
+import SearchBar from '../SearchBar';
+
+
+
+const NavBar = () => {
+  const classes = useNavBarStyles();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const myState = useSelector((state) => state.authReducer);
+  const [redirect, setRedirect] = useState(false);
+
+  const handleLogOut = () => {
+      dispatch(setToken({
+        access_token:null,
+        token_type:null,
+        scope: null
+      }));
+      localStorage.removeItem("tokenConfig");
+      dispatch(setLikedPhotosId([],1));
```

Comment:
Instead of dispatching this action from here we can dispatch if from the saga if we have that flow. Or we can have a logout action that does all of the above work:
- Clearing auth token.
- Removing token from local storage
- Resetting all state data